js/widgets/phrasemaker.js

```diff
@@ -1464,6 +1419,42 @@ class PhraseMaker {
                 this.rowLabels[blockIndex] + ": " + this.rowArgs[blockIndex];
         };
 
+        const __enterArgValue = () => {
+            this.blockValue = this._pitchWheel.navItems[
+                this._pitchWheel.selectedNavItemIndex
+            ].title;
+            docById("wheelnav-_exitWheel-title-1").children[0].textContent = this.blockValue;
+            __selectionChanged(true);
+        };
+
+        if (condition === "graphicsblocks") {
+            if (blockLabel === "forward" || blockLabel === "back") {
+                for (let i = 0; i < forwardBackLabel.length; i++) {
+                    this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+                }
+            } else if (blockLabel === "right" || blockLabel === "left") {
+                for (let i = 0; i < leftRightLabel.length; i++) {
+                    this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+                }
+            } else if (blockLabel === "setheading") {
+                for (let i = 0; i < setHeadingLabel.length; i++) {
+                    this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+                }
+            } else if (blockLabel === "setpensize") {
+                for (let i = 0; i < setPenSizeLabel.length; i++) {
+                    this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+                }
+            } else {
+                for (let i = 0; i < setLabel.length; i++) {
+                    this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+                }
+            }
+        } else if (condition === "synthsblocks") {
+            for (let i = 0; i < valueLabel.length; i++) {
+                this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
+            }
+        }
```

Comment:
Please make this code dry. 

Declare a variable => `labelLength` 
put if else according to your conditions and set value of `noteStored`

Use for loop once.
Something like:- 

```
let labelLength = 0;
if (condition === "graphicsblocks") {
            if (blockLabel === "forward" || blockLabel === "back") {
                labelLength = forwardBackLabel.length
            }
            } else if (blockLabel === "right" || blockLabel === "left") {
                labelLength = leftRightLabel.length;
               
               ...
            //loop just once
            for (let i = 0; i < labelLength; i++) {
                this._pitchWheel.navItems[i].navigateFunction = __enterArgValue;
            }
        }

Reply:
Yeah, sure.